public/app/features/dashboard-scene/serialization/sceneVariablesSetToVariables.ts

```diff
@@ -331,27 +332,30 @@ export function sceneVariablesSetToSchemaV2Variables(
         dataQuery = {
           kind: 'DataQuery',
           version: defaultDataQueryKind().version,
-          group: datasource?.type ?? getDataQueryKind(query),
+          group: datasource?.type || getDataQueryKind(query),
           ...(datasource?.uid && {
             datasource: {
               name: datasource.uid,
             },
           }),
-          spec: getDataQuerySpec(query),
+          spec: query,
         };
       } else {
+        // Only include LEGACY_STRING_VALUE_KEY if query is a non-empty string
+        const spec: Record<string, string> = {};
+        if (query) {
+          spec[LEGACY_STRING_VALUE_KEY] = query;
```

Comment:
The legacy string value is the old way for variable queries to define a query instead of an object. We keep support through this `LEGACY_STRING_VALUE_KEY`. This was already existing, we're just considering it when serializing V2